pkg/cluster/internal/kubeadm/config.go

```diff
@@ -107,6 +107,10 @@ type DerivedConfigData struct {
 	IPv6 bool
 	// kubelet cgroup driver, based on kubernetes version
 	CgroupDriver string
+	// JoinSkipPhases are the skipPhases values for the JoinConfiguration.
+	JoinSkipPhases []string
+	// InitSkipPhases are the skipPhases values for the InitConfiguration.
+	InitSkipPhases []string
```

Comment:
was this discussed on the ticket?
kind doesn't have to wrap these kubeadm options, and users can pass a custom slice via kind kubeadm config patches, which would override the default to skip the preflight.

Reply:
Not sure we got this deep into detail in the issue beyond setting `skipPhases` in the config (https://github.com/kubernetes-sigs/kind/issues/3606#issuecomment-2123157692), but I can see where we could get rid of these fields by hardcoding `preflight` and conditionally adding `addon/kube-proxy` right in the template. Is that what you mean? Or are you suggesting not to specify any `skipPhases` by default? Overall I was trying to preserve the existing default behavior since I don't fully understand the implications of running the preflight checks here.

Reply:
NVM, i mistakenly thought that this exposes the skip phase slices in the public facing kind config, which is not the case.

Reply:
yeah, naming is hard and all that, this is internal / plumbing options to the config templates